### PR TITLE
Web server in rust runtime

### PR DESCRIPTION
**Pull request by dfellis**
Resolves \#233 and also cleans up the string manipulation logic in the opcodes.

### CODE REVIEW

File: runtime/src/vm/opcode.rs [hunk 1]
```diff
@@ -1,25 +1,38 @@
...
 use tokio::time::delay_for;
 use twox_hash::XxHash64;
 
-use crate::vm::event::{EventEmit, HandlerFragment};
+use crate::vm::event::{BuiltInEvents, EventEmit, HandlerFragment};
 use crate::vm::instruction::InstructionScheduler;
 use crate::vm::memory::{CLOSURE_ARG_MEM_START, HandlerMemory};
+use crate::vm::run::{EVENT_TX};
+
+static HTTP_RESPONSES: Lazy<Arc<Mutex<HashMap<i64, HandlerMemory>>>> = Lazy::new(|| {
```
Comment (depombo) [inline runtime/src/vm/opcode.rs:33]: why is a Mutex necessary, wouldn't a RwLock suffice?

File: runtime/src/vm/opcode.rs [hunk 2]
```diff
@@ -3074,110 +2560,188 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
...
+    let event_tx = EVENT_TX.get().unwrap();
+    if event_tx.send(event_emit).is_err() {
+      eprintln!("Event transmission error");
+      std::process::exit(3);
+    }
+    // Get the HTTP responses lock and periodically poll it for responses from the user code
+    // TODO: Swap from a timing-based poll to getting the waker to the user code so it can be
+    // woken only once and reduce pressure on this lock.
+    let responses = Arc::clone(&HTTP_RESPONSES);
+    let response_hm = poll_fn(|cx: &mut Context<'_>| -> Poll<HandlerMemory> {
```
Comment (depombo) [inline runtime/src/vm/opcode.rs:2668]: oh rust..

File: runtime/src/vm/opcode.rs (hunk 1 above)
Comment (dfellis) [reply runtime/src/vm/opcode.rs:33]: https://stackoverflow.com/questions/50704279/when-or-why-should-i-use-a-mutex-over-an-rwlock#50704283

I am pretty sure HandlerMemory needs the extra safeguards. I am willing to be proven wrong here, though. I am also expecting us to eventually see this as a bottleneck and switch to dashmap or something like that.

File: runtime/src/vm/opcode.rs (hunk 2 above)
Comment (dfellis) [reply runtime/src/vm/opcode.rs:2668]: I am glad I found that thing, though! I wasn't sure how I was going to implement this without something of this nature.

And I should be able to make it more efficient (and not really a poll at all) in the future.

We could also have timeout support at some point to make sure that there aren't dead connections piling up.

**Pull request merged**